### PR TITLE
expression: remove the usage of "TypeClass" in "builtin_op.go"

### PR DESCRIPTION
**Pull request by zz-jason**
to #4394

### CODE REVIEW

File: expression/builtin_op.go [hunk 1]
```diff
@@ -339,13 +339,16 @@ func (c *isTrueOrFalseFunctionClass) getFunction(ctx context.Context, args []Exp
 		return nil, errors.Trace(err)
 	}
 
-	argTp := tpInt
```
Comment (XuHuaiyu) [inline expression/builtin_op.go]: this is an equivalent replacement?
should we keep it here, and use `fieldTp2EvalTp` in line 343?

Comment (zz-jason) [reply expression/builtin_op.go]: yes, `ClassReal` == `tpReal` and `ClassDecimal` == `tpDecimal`

File: expression/builtin_op.go [hunk 2]
```diff
@@ -678,25 +678,7 @@ func (c *isNullFunctionClass) getFunction(ctx context.Context, args []Expression
...
-		tp := args[0].GetType().Tp
-		if types.IsTypeTime(tp) {
-			argTp = tpDatetime
-		} else if tp == mysql.TypeDuration {
-			argTp = tpDuration
-		} else {
-			argTp = tpString
-		}
-	}
+	argTp := fieldTp2EvalTp(args[0].GetType())
```
Comment (XuHuaiyu) [inline expression/builtin_op.go]: We should check tpDatetime and tpTimestamp here,
if tpDatetime || tpTimestamp {
    argTp = tpDatetime
}

File: expression/builtin_op.go [hunk 3]
```diff
@@ -549,19 +552,16 @@ func (c *unaryMinusFunctionClass) handleIntOverflow(arg *Constant) (overflow boo
...
-	case types.ClassDecimal:
-		tp = tpDecimal
 	}
 
 	sc := ctx.GetSessionVars().StmtCtx
 	overflow := false
 	// TODO: Handle float overflow.
 	if arg, ok := argExpr.(*Constant); sc.InSelectStmt && ok &&
-		arg.GetTypeClass() == types.ClassInt {
+		fieldTp2EvalTp(arg.GetType()) == tpInt {
```
Comment (XuHuaiyu) [inline expression/builtin_op.go]: we can use `tp == tpInt` here,
since `argExpr.GetType() == arg.GetType()`

**Comment by coocood** [comment]
LGTM

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

**Pull request merged**